### ISSUE TITLE
Bazel crashes with NPE

### ISSUE THREAD

**Issue by tgeng** [bazelbuild/bazel#23483]
### Description of the bug:

We noticed the following error occasionally in our CI environment.
```
[2024-08-30T22:44:39.669Z] + bazel build --local_resources=memory=69578 --local_resources=cpu=10 --jobs=10 --config=ci --stamp ... (target patterns are omitted for confidentiality)
[2024-08-30T22:44:39.669Z] Starting bazel-event-subscriber as a background process...
[2024-08-30T22:44:39.669Z] Starting local Bazel server and connecting to it...
[2024-08-30T22:44:48.060Z] INFO: Invocation ID: 23da8df7-70a1-4c83-8d8a-e2bd6dec2c63
[2024-08-30T22:44:48.060Z] Computing main repo mapping: 
[2024-08-30T22:44:48.980Z] Computing main repo mapping: 
[2024-08-30T22:44:49.901Z] Computing main repo mapping: 
[2024-08-30T22:44:51.257Z] Computing main repo mapping: 
[2024-08-30T22:44:51.815Z] Loading: 
[2024-08-30T22:44:51.815Z] Loading: 2 packages loaded
[2024-08-30T22:44:53.170Z] Analyzing: nullFATAL: bazel crashed due to an internal error. Printing stack trace:
[2024-08-30T22:44:53.170Z] java.lang.NullPointerException: Cannot invoke "String.isEmpty()" because "message" is null
[2024-08-30T22:44:53.170Z] 	at com.google.devtools.build.lib.runtime.SkymeldUiStateTracker.writeLoadingAnalysisPhaseProgress(SkymeldUiStateTracker.java:147)
[2024-08-30T22:44:53.170Z] 	at com.google.devtools.build.lib.runtime.SkymeldUiStateTracker.writeProgressBar(SkymeldUiStateTracker.java:92)
[2024-08-30T22:44:53.170Z] 	at com.google.devtools.build.lib.runtime.UiEventHandler.addProgressBar(UiEventHandler.java:1017)
[2024-08-30T22:44:53.170Z] 	at com.google.devtools.build.lib.runtime.UiEventHandler.doRefresh(UiEventHandler.java:878)
[2024-08-30T22:44:53.170Z] 	at com.google.devtools.build.lib.runtime.UiEventHandler.lambda$startUpdateThread$0(UiEventHandler.java:954)
[2024-08-30T22:44:53.170Z] 	at java.base/java.lang.Thread.run(Unknown Source)
[2024-08-30T22:44:53.170Z] 
[2024-08-30T22:44:53.170Z] Server terminated abruptly (error code: 14, error message: 'Socket closed', log file: '/mnt/jenkins/.cache/bazel/_bazel_jenkins/34b69d59fa4fc66fd3c84f6db0658cf9/server/jvm.out')
```

### Which category does this issue belong to?

Core

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

We don't know how to reliably reproduce this. This happens a couple of times in a week, during which we run thousands of builds so it's pretty rare.

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

release 7.3.1

**Comment by benjaminp** [comment]
4972db8a5211c2efd7af3a3b77f24f7698e4bdfe

**Comment by tgeng** [comment]
Thanks! Will this be cherry picked to bazel 7.4?

**Comment by fmeum** [comment]
@bazel-io fork 7.4.0

**Comment by meisterT** [comment]
cc @joeleba

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!
